src/gc.c

```diff
@@ -2712,6 +2723,45 @@ DLLEXPORT void *jl_gc_counted_realloc_with_old_size(void *p, size_t old, size_t
     return b;
 }
 
+DLLEXPORT void *jl_malloc(size_t sz)
+{
+    size_t *p = (size_t *)jl_gc_counted_malloc(sz + sizeof(size_t));
+    p[0] = sz;
+    return (void *)(p + 1);
```

Comment:
we might as well have this be 16 aligned then. I would think that libraries actually needing such alignment would be defensive about it, but let's not be greedy for 8 bytes (aka the Jameson way :p )